### PR TITLE
Add tokens parameter to solana-genesis

### PR DESCRIPTION
**Pull request by CriesofCarrots**

### CODE REVIEW

File: multinode-demo/setup.sh [hunk 1]
```diff
@@ -87,7 +87,7 @@ if $node_type_leader; then
   $solana_mint <<<"$num_tokens" > "$SOLANA_CONFIG_PRIVATE_DIR"/mint.json
 
   echo "Creating $SOLANA_CONFIG_DIR/genesis.log"
-  $solana_genesis < "$SOLANA_CONFIG_PRIVATE_DIR"/mint.json > "$SOLANA_CONFIG_DIR"/genesis.log
+  $solana_genesis --tokens "$num_tokens" < "$SOLANA_CONFIG_PRIVATE_DIR"/id.json > "$SOLANA_CONFIG_DIR"/genesis.log
```
Comment (garious) [inline multinode-demo/setup.sh]: This was actually the only correct spot to call it `mint.json`. Can you revert?

File: src/bin/genesis.rs [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
+            Arg::with_name("tokens")
+                .short("t")
+                .long("tokens")
+                .value_name("NUMBER")
+                .takes_value(true)
+                .required(true)
+                .help("Number of tokens with which to initialize mint"),
+        )
+        .get_matches();
+    let tokens: i64;
```
Comment (garious) [inline src/bin/genesis.rs]: Clippy won't like this.

Comment (garious) [reply src/bin/genesis.rs]: Weird, I checked the log. No complaint from clippy. It generally prefers:

```rust
let tokens = if { foo } else {bar};
```

Comment (CriesofCarrots) [reply src/bin/genesis.rs]: I imagine it's because of the `if let` construction in the following expression...?

**Comment by garious** [comment]
Yoinked by #597

**Pull request closed without merging**